### PR TITLE
[Linter] Set up linter

### PR DESCRIPTION
**Pull request by luongt13**
This PR adds a linter (husky, eslint, prettier). Linter will run before commits.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+node_modules
+public
+build
+src/locales
+
+// ignore all test files
+*.test.js
```
Comment (briannapoulos) [inline .eslintignore:7]: I think it'd be good to have new lines at the end of these files

File: src/App.js [hunk 2]
```diff
@@ -71,87 +71,98 @@ const theme = {
...
+              )}
+              <a
+                style={{ color: '#0d47a1' }}
+                href="https://app.crosscut.io/"
+                target="_blank"
+                rel="noreferrer"
+              >
+                app.crosscut.io
+              </a>
+              .
```
Comment (briannapoulos) [inline src/App.js:139]: This looks hard to read -- could the text be wrapped in the code without wrapping it in the display?

File: src/components/Info/Info.jsx [hunk 3]
```diff
@@ -1,70 +1,191 @@
...
-                <p>{i18n.t("Go to")} <a style={{ color: '#0d47a1'}} href="https://app.crosscut.io/" target="_blank" >app.crosscut.io</a></p>
-                <Divider/>
-                <ModalTitle>{i18n.t("Contact Us")}</ModalTitle>
-                <p>{i18n.t("Having trouble? Have ideas for new features?")}</p>
-                <p>{i18n.t("Email us at")} <a style={{ color: '#0d47a1' }} href="mailto:coite@crosscut.io" target="_blank">coite@crosscut.io</a></p>
-            </div>
-            }
-        </ModalContent>
-        <ModalActions><ButtonItem handleClick={setShowInfoModal} buttonText={i18n.t("Close")} primary={true}/></ModalActions>
+  // TO-DO: refactor this code to improve how we're handling text
```
Comment (briannapoulos) [inline src/components/Info/Info.jsx:29]: Let's change this to `TODO`
```suggestion
  // TODO: refactor this code to improve how we're handling text
```

File: package.json [hunk 4]
```diff
@@ -14,10 +14,17 @@
     "cluster:clean": "d2 cluster down 2.38 --clean",
     "test": "d2-app-scripts test",
     "deploy": "d2-app-scripts deploy",
-    "format": "prettier . -w"
+    "format": "prettier --write ./src",
+    "lint": "eslint --fix ./src",
+    "prepare": "husky install"
```
Comment (briannapoulos) [inline package.json:19]: Do you think it makes sense to add a github action that will do a lint test on any PR and fail if the linting fails?

File: .prettierrc.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+    "trailingComma": "es5",
+    "tabWidth": 2,
+    "semi": true,
+    "singleQuote": true
+  }
+
```
Comment (briannapoulos) [inline .prettierrc.json:7]: It looks like this is still showing as missing a newline

File: .husky/pre-commit [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
```
[the remark this answers is not in the document]
Comment (luongt13) [reply .husky/pre-commit:2]: yes, I've added `"prefer-const": 2`

**Review by briannapoulos — APPROVED** [review]
I think it looks good. There are two outstanding comments (github actions and wrapping the really long line). I would make sure to test at least the basic functionality before deploying, since it touches so much of the code.

File: package.json (hunk 4 above)
Comment (luongt13) [reply package.json:19]: Will add github action in a separate PR.

**Pull request merged**